### DISCUSSION TITLE
[Enhancement] Add interfaces for all public APIs to make mocking easier

### DISCUSSION THREAD

**Discussion by johnshardman** [dotnet/maui#56] [General] [answered]
## Summary
To make it easier to unit test view (the V in MVVM) code, add interfaces for all public APIs. For example, have Button implement IButton. That way, those who want to be able to unit test their view code can avoid using concrete classes, and can mock IButton in unit tests.

## API Changes

All public APIs to implement interfaces.

## Intended Use Case
For use in unit testing of view code.

**Comment by dhindrik** [comment, marked as answer]
It looks like it is a part of the Slim Renderer Architecture, https://github.com/dotnet/maui/issues/28
